### ISSUE TITLE
Code review for Pijul Nest

### ISSUE THREAD

**Issue by notramo** [KB1RD/matrix-notepad#19]
[Pijul](https://pijul.org/) is a patch based version control system with better conflict resolution than Git.
There is a social coding platform for Pijul: https://nest.pijul.org/
At the moment there is no code review feature in the Pijul Nest.
Is there any way it could be implemented in Matrix Notepad?
This would require #14 to be completed.
Maybe some server side code would be required (to fetch the patches from the Pijul Nest).
The original content of the patch can't be modified, so every edit should be rendered as a suggestion.
Would this be useful in Matrix Notepad, or it would be feature creep?

**Comment by KB1RD** [comment]
@notramo I don't think that this is something that makes sense to implement in Matrix Notepad directly.
***BUT...***
Matrix Notepad is built on top of Matrix and uses an event-based protocol that can be "spoken" by any program that can receive events from the Matrix server hosting the account. Matrix introduces the concept of [bridges](https://matrix.org/bridges/), which are designed to allow users from another chat platform to chat with users on Matrix. A bridge could be created that bridges edit events instead of chat messages using the same concept. Using this, the bridge (which would pose as a bot) could create a room and send insertion events for the remote patches. This would allow editing collaboratively through Matrix Notepad.

You mentioned that the original patch cannot be modified. Since Matrix Notepad does not redact old data, the original patch would not be deleted, only not rendered by the client. This is not desirable, though. I am working on multi-branch support as well as per-user branches. The bridge could make it so that all the users are forced into a suggestion branch. Again, the support is not present *yet*, but that's in the works.

I'll keep the issue open because it should be possible in the future. Bridging is something that I intend to support and I'm excited to see that there is a use for it!
